zetaclient/chains/solana/rpc/rpc.go

```diff
@@ -122,6 +127,25 @@ func GetSignaturesForAddressUntil(
 	return allSignatures, nil
 }
 
+// GetTransactionWithMaxVersion fetches a transaction with the given signature and max version.
+func GetTransactionWithMaxVersion(
+	ctx context.Context,
+	client interfaces.SolanaRPCClient,
+	signature solana.Signature,
+	maxTxVersion *uint64,
+) (*rpc.GetTransactionResult, error) {
+	txResult, err := client.GetTransaction(ctx, signature, &rpc.GetTransactionOpts{
+		MaxSupportedTransactionVersion: maxTxVersion,
+	})
+
+	// skip unsupported transaction version error
+	if err != nil && strings.Contains(err.Error(), ErrorCodeUnsupportedTransactionVersion) {
+		return nil, nil
```

Comment:
Can we rather return a boolean unsupported, or a error that can be filtered? I think would be cleaner than returning a nil

Reply:
Returning an additional boolean `skip` looks more readable.

Reply:
added boolean https://github.com/zeta-chain/node/pull/3206/commits/4f0e1716e43c045e4a6427f0df2738e231008429